### PR TITLE
Add feature of getting root folder. Refactor actions.

### PR DESCRIPTION
**Pull request by kuzmavictor**

### CODE REVIEW

File: app/service/store/action/index.js [hunk 1]
```diff
@@ -1,22 +1,60 @@
...
   removeItem: (store) => (id) => {
-    const promise = store.requestServiceProvider.fileListService.removeFile(id);
-    promise.then((response) => {
-      store.commit('removeItem', id);
-      store.dispatch('fetchFiles');
-    });
+    store.commit('setProcessStatus', ITEM_REMOVING_MSG);
+    return store.requestServiceProvider.fileListService.removeFile(id)
+        .then((status) => {
+          store.commit('removeItem', id);
```
Comment (dkalinin) [inline app/service/store/action/index.js]: Please remove this mutation.

File: app/service/store/action/index.js [hunk 2]
```diff
@@ -26,38 +64,72 @@ export const actions = {
...
-    store.commit('isFileListLoading', true);
-    const promise = store.requestServiceProvider.fileListService.getFiles('root');
-    promise.then((response) => {
-      store.commit('fetchFiles', store.state.files);
-      store.commit('isFileListLoading', false);
-    });
+    store.commit('setProcessStatus', FILES_FETCHING_MSG);
+    return store.requestServiceProvider.fileListService.getFiles(store.state.currentFolder)
+        .then((fileList) => {
+          store.commit('fetchFiles', store.state.files);
```
Comment (dkalinin) [inline app/service/store/action/index.js]: fetchFiles => setFiles

File: app/service/store/action/index.js [hunk 3]
```diff
@@ -26,38 +64,72 @@ export const actions = {
...
+        .finally(()=>{
+          store.commit('setProcessStatus', '');
+        });
   },
+
   /**
    * Adds files to file list. The promise that returns emulates working with server-side.
    *
    * @param {Store} store - an instance of {@link Store}
    * @return {function([],int)}
```
Comment (dkalinin) [inline app/service/store/action/index.js]: Promise

File: app/service/store/action/index.js [hunk 4]
```diff
@@ -26,38 +64,72 @@ export const actions = {
...
-        .fileListService.addFiles(fileList, folderId);
-    promise.then((response) => {
-      store.commit('addFiles', fileList, folderId);
-      store.dispatch('fetchFiles');
-    });
+    store.commit('setProcessStatus', FILES_ADDING_MSG);
+    return store.requestServiceProvider
+        .fileListService.addFiles(fileList, folderId)
+        .then((files) => {
+          store.commit('addFiles', fileList, folderId);
```
Comment (dkalinin) [inline app/service/store/action/index.js]: This mutation is not needed.

File: app/service/store/action/index.js [hunk 5]
```diff
@@ -26,38 +64,72 @@ export const actions = {
...
-   * @return {Promise<any>}
+   * @return {function(int)}
    */
   getShareableUrl: (store) => (id) => {
-    return store.requestServiceProvider.fileListService.getShareableUrl(id)
-        .then((response)=> {
-          return (response.body);
+    return store.requestServiceProvider.fileListService.getShareUrl(id)
+        .then((shareableUrl) => {
+          return (shareableUrl);
```
Comment (dkalinin) [inline app/service/store/action/index.js]: Please remove ().

File: app/service/store/action/index.js [hunk 6]
```diff
@@ -26,38 +64,72 @@ export const actions = {
...
+          store.commit('setProcessStatus', '');
+        });
   },
+
   /**
    * Gets a shareable URL from server side.
    *
    * @param {Store} store - an instance of {@link Store}
-   * @return {Promise<any>}
+   * @return {function(int)}
```
Comment (dkalinin) [inline app/service/store/action/index.js]: Promise

File: app/service/store/action/index.js [hunk 7]
```diff
@@ -26,38 +64,72 @@ export const actions = {
...
+   * Moves a file or a folder to other folder.
+   *
+   * @param {Store} store - an instance of {@link Store}
+   * @return {function([],int)}
+   */
+  moveFile: (store) => (file, folderId) => {
+    store.commit('setProcessStatus', FILE_MOVING_MSG );
+    return store.requestServiceProvider.fileListService.moveFile(file, folderId)
+        .then((fileModel) => {
+          store.commit('setLoadingProcess', '');
```
Comment (dkalinin) [inline app/service/store/action/index.js]: setProcessStatus

File: app/service/store/action/index.js [hunk 8]
```diff
@@ -26,38 +64,72 @@ export const actions = {
...
+          store.dispatch('fetchFiles', store.state.files);
+        })
+        .catch((error) => {
+          store.commit('setErrorMessage', error);
```
Comment (dkalinin) [inline app/service/store/action/index.js]: setProcessStatus

File: app/view/component/file-list/index.js [hunk 9]
```diff
@@ -15,7 +15,8 @@ export class FileList extends DraggableComponent {
    */
   constructor(parentRoot, store) {
     super(parentRoot, store);
-    this._store.events.subscribe('stateChange', () => this._render());
+    this._store.events.subscribe('stateChange.statusOperation', () => this._render());
```
Comment (dkalinin) [inline app/view/component/file-list/index.js]: Just change text in loading span.

**Comment by kuzmavictor** [comment]
@dkalinin fixed, PTAL.

File: app/view/component/file-list/index.js [hunk 10]
```diff
@@ -65,11 +76,55 @@ export class FileList extends DraggableComponent {
...
+   * Shows an error message if fetching files is failed.
+   *
+   * @private
+   */
+  _showFetchingFileErrorMessage() {
+    if (typeof this._fetchingFilesError === 'undefined') {
+      this._fetchingFilesError = new Span(this._root);
+      this._fetchingFilesError.text = this.state.fetchingFilesError;
+    } else {
+      this._fetchingFilesError.text = this.state.fetchingFilesError;
```
Comment (dkalinin) [inline app/view/component/file-list/index.js]: Please remove duplicated code.

File: app/view/component/file-list/index.js [hunk 11]
```diff
@@ -65,11 +76,55 @@ export class FileList extends DraggableComponent {
...
+   * Shows an error message if getting root folder is failed
+   *
+   * @private
+   */
+  _showGettingRootFolderErrorMessage() {
+    if (typeof this._gettingRootFolderError === 'undefined') {
+      this._gettingRootFolderError = new Span(this._root);
+      this._gettingRootFolderError.text = this.state.gettingRootFolderError;
+    } else {
+      this._gettingRootFolderError.text = this.state.gettingRootFolderError;
```
Comment (dkalinin) [inline app/view/component/file-list/index.js]: Please remove duplicated code.

File: app/view/component/file-list/index.js [hunk 12]
```diff
@@ -65,11 +76,55 @@ export class FileList extends DraggableComponent {
...
    */
-  _createLoadingSpan() {
-    this._loadingSpan = new Span(this._root);
-    this._loadingSpan.text = 'Please wait...';
+  _showOperationStatus() {
+    if (typeof this._loadingStatus === 'undefined') {
+      this._loadingStatus = new Span(this._root);
+      this._loadingStatus.text = this.state.statusOperation;
+    } else {
+      this._loadingStatus.text = this.state.statusOperation;
```
Comment (dkalinin) [inline app/view/component/file-list/index.js]: Please remove duplicated code.

File: app/view/component/page/file-list/index.js [hunk 13]
```diff
@@ -23,6 +25,7 @@ export class FileListPage extends Component {
    * @private
    */
   _render() {
+    this._errorFileList = new Span(this._root);
```
Comment (dkalinin) [inline app/view/component/page/file-list/index.js]: This is not used.

**Comment by kuzmavictor** [comment]
@dkalinin fixed, PTAL.

- the feature to handle errors of adding files will be added in https://github.com/kuzmavictor/FileStorageUI/pull/14

File: app/service/store/action/index.js [hunk 14]
```diff
@@ -21,7 +21,8 @@ export const actions = {
     store.commit('setProcessStatus', CONNECTION_MSG);
     return store.requestServiceProvider.fileListService.getRootFolder()
         .then((fileModel) => {
```
Comment (dkalinin) [inline app/service/store/action/index.js]: It is more descriptive to have `folder` as parameter instead of `fileModel`.

**Pull request merged**